From 995780bedee2b5f8809df684553c0045d4b68f71 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Fri, 1 Aug 2008 15:08:44 +0800
Subject: [PATCH] ENGR00086209 MX35 can not bootup due to IPU reset codes

Move the IPU_CONF reset below the ipu core clock enable

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 drivers/mxc/ipu/ipu_common.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mxc/ipu/ipu_common.c b/drivers/mxc/ipu/ipu_common.c
index 3b1f696..089d0dd 100644
--- a/drivers/mxc/ipu/ipu_common.c
+++ b/drivers/mxc/ipu/ipu_common.c
@@ -99,9 +99,6 @@ int ipu_probe(struct platform_device *pdev)
 	g_ipu_dev = &pdev->dev;
 	g_ipu_hw_rev = ipu_conf->rev;
 
-	/* resetting the CONF register of the IPU */
-	__raw_writel(0x00000000, IPU_CONF);
-
 	/* Register IPU interrupts */
 	g_ipu_irq[0] = platform_get_irq(pdev, 0);
 	if (g_ipu_irq[0] < 0)
@@ -132,6 +129,9 @@ int ipu_probe(struct platform_device *pdev)
 
 	clk_enable(g_ipu_clk);
 
+	/* resetting the CONF register of the IPU */
+	__raw_writel(0x00000000, IPU_CONF);
+
 	__raw_writel(0x00100010L, DI_HSP_CLK_PER);
 
 	/* Set SDC refresh channels as high priority */
-- 
1.5.4.4

